src/cnv/ugbio_cnv/convert_combined_cnv_results_to_vcf.py

```diff
@@ -0,0 +1,183 @@
+                record.filter.add("UG-CNV-LCR")
+            else:
+                record.filter.add("PASS")
+            if isinstance(copy_number_value, float):
+                record.info["CopyNumber"] = copy_number_value
+            if isinstance(copy_number_value, float):
+                record.info["RoundedCopyNumber"] = int(round(copy_number_value))
+            record.info["SVLEN"] = int(end) - int(start)
+            record.info["SVTYPE"] = cnv_type
+            record.info["CNV_SOURCE"] = cnv_call_source
+            # END position is automatically generated for multi-base variants
+
+            # Set genotype information for each sample
+            gt = [None, 1]
+            if isinstance(copy_number_value, float):
+                if int(round(copy_number_value)) == 1:
+                    gt = [0, 1]
+                elif int(round(copy_number_value)) == 0:
+                    gt = [1, 1]
+            record.samples[sample_name]["GT"] = (gt[0], gt[1])
+
+            # Write the record to the VCF file
+            vcf_out.write(record)
+
+
+def run(argv):
+    """
+    converts combined CNV calls (from cnmops, cnvpytor, gridss) in bed format to vcf.
+    input arguments:
+    --cnv_annotated_bed_file: input bed file holding CNV calls.
+    --fasta_index_file: (.fai file) tab delimeted file holding reference genome chr ids with their lengths.
+    --out_directory: output directory
+    --sample_name: sample name
+    output files:
+    vcf file: <sample_name>.cnv.vcf.gz
+        shows called CNVs in zipped vcf format.
+    vcf index file: <sample_name>.cnv.vcf.gz.tbi
+        vcf corresponding index file.
+    """
+    parser = argparse.ArgumentParser(
+        prog="convert_combined_cnv_results_to_vcf.py", description="converts CNV calls in bed format to vcf."
+    )
+
+    parser.add_argument("--cnv_annotated_bed_file", help="input bed file holding CNV calls", required=True, type=str)
+    parser.add_argument(
+        "--fasta_index_file",
+        help="tab delimeted file holding reference genome chr ids with their lengths. (.fai file)",
+        required=True,
+        type=str,
+    )
+    parser.add_argument("--out_directory", help="output directory", required=False, type=str)
+    parser.add_argument("--sample_name", help="sample name", required=True, type=str)
+    parser.add_argument("--verbosity", help="Verbosity: ERROR, WARNING, INFO, DEBUG", required=False, default="INFO")
+
+    args = parser.parse_args(argv[1:])
+    logger.setLevel(getattr(logging, args.verbosity))
+
+    header = add_vcf_header(args.sample_name, args.fasta_index_file)
+
+    # Open a VCF file for writing
+    if args.out_directory:
+        out_directory = args.out_directory
+    else:
+        out_directory = ""
+    outfile = pjoin(out_directory, args.sample_name + ".cnv.vcf.gz")
+    write_vcf(outfile, header, args.cnv_annotated_bed_file, args.sample_name)
+
+    # index outfile
+    try:
+        cmd = ["bcftools", "index", "-t", outfile]
```

Comment:
Huh? Why can this fail?

Reply:
I actually like this checkup for vcf validity. 
if somehow the output vcf is not valid, it will not be able to index it and the user will know about it.